### ISSUE TITLE
Disable SMB?

### ISSUE THREAD

**Issue by botherder** [securitywithoutborders/hardentools#24]
In the light of WannaCry, I've been wondering if it would make sense to add an option to disable SMBv1.
Thoughts?

**Comment by ccoenen** [comment]
this reddit thread seems related:
https://www.reddit.com/r/sysadmin/comments/6baquq/what_still_uses_smbv1/

IMO, SMBv1 should be deactivated.

**Comment by deadbits** [comment]
IMHO, I think it's worth disabling even if it wasn't for WannaCry. On Windows 10 (not sure about Win8), if you are trying to manually perform a task that requires file sharing via SMB to be turned on a little prompt appears in File Explorer asking if you want to adjust your settings. Point being, the user could always manually re-enable it that use case if they actually _need_ it.

When setting up a fresh Windows system for personal / daily use, disabling SMB is one of the first things I do anyways.
It does cause problems with things like a local NAS, of course, but in those situations I just enable it temporarily and then disable. 

Perhaps add the capability and warn users of the possible side effects like that^? ```Disabling SMB will restrict file sharing within your local network... If you manually enable it if prompted to do so, you will need to run HardenTools again to disable it.``` or something similar to that?

**Comment by stuchl4n3k** [comment]
Totally agree with SMBv1 deactivation.

Regarding any warning or info - HT already disables a lot of "features", so I don't see a reason to bother folks with sth. like that just because of file sharing. 

On the other hand, some overview of disabled/impacted features would be nice, but that's for another feature request...

**Comment by Velocet** [comment]
It only disables SMBv1 (1.0/CIFS)

2.0 was introduced with Vista. So i don't think that you have to give any warning at all.

**Comment by obsti8383** [comment]
Did a first version to disable SMBv1 (needs improvement regarding revert to original state) -> https://github.com/obsti8383/hardentools/tree/smbv1 (smbv1.go is the central source).

But this really breaks things, e.g. I can't access a shared folder on my fritz box anymore (beneath it beeing mapped as an drive). I think this could also be relevant for our intended users, since there might still be some NAS boxes/Routers out there.

I would only include that as optional/in expert mode after we extendend hardentools GUI to allow that, but open on discussions on that. (Could you give it a try and tell me if you experience similar problems? Please be aware that the build script for this branch only works on 64bit OSes right now)

**Comment by ccoenen** [comment]
I just disabled smbv1 manually here, and you're right. It breaks a lot of stuff. I own a 2012 QNAP NAS (which is still getting software support), but apprently it needs SMBv1.
I tried their settings according to [SMBv3 on QTS 4.2](https://www.qnap.com/en/how-to/tutorial/article/how-to-use-smb-3-0-in-qts-4-2) but I don't have those settings (yes, I _do_ have QTS 4.2). So I guess my NAS is limited to SMBv1?

Long story short: you're right, it may break more than I'd have thought.

![2017-07-23 09_32_48-twitter](https://user-images.githubusercontent.com/124909/28497593-70c7e742-6f8b-11e7-9284-5a8a93d1bf0d.png)

**Comment by botherder** [comment]
Shall we close this then?

**Comment by deadbits** [comment]
Also tested on my NAS after seeing these messages... Disabling it breaks my access to my NAS as well.  I unfortunately vote for closing this since it apparently will break a lot more than initially thought :/
